server/sonar-server/src/main/java/org/sonar/server/plugins/PluginCompression.java

```diff
@@ -50,12 +50,25 @@ public PluginCompression(Configuration configuration) {
     this.configuration = configuration;
   }
 
-  public void compressJar(String pluginKey, Path jarFile) {
+  public void compressJar(String pluginKey, Path sourceDir, Path targetJarFile) {
     if (configuration.getBoolean(PROPERTY_PLUGIN_COMPRESSION_ENABLE).orElse(false)) {
-      Path pack200Path = FileUtils.getPack200FilePath(jarFile);
-      pack200(jarFile, pack200Path, pluginKey);
-      String hash = calculateMd5(pack200Path);
-      RemotePluginFile compressedPlugin = new RemotePluginFile(pack200Path.getFileName().toString(), hash);
+      Path targetPack200Path = FileUtils.getPack200FilePath(targetJarFile);
+      Path sourcePack200Path = sourceDir.resolve(targetPack200Path.getFileName());
+
+      // check if packed file was deployed alongside the jar. If that's the case, use it instead of generating it (SONAR-10395).
+      if (Files.isRegularFile(sourcePack200Path)) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Replace this with a call to the "toFile().isFile()" method ([squid:S3725](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3725))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Fcompress_plugins&issues=AWF1-O5v0orRW2HVTxyO&open=AWF1-O5v0orRW2HVTxyO)

[ISSUE_KEY]: AWF1-O5v0orRW2HVTxyO